cogs/seasonstats.py

```diff
@@ -521,6 +519,129 @@ def __init__(self, bot):
+        """
+        season = season or await self.bot.seasonconfig.get_season_id() - 1
+        query = """SELECT player_tag, end_defenses - start_defenses as defenses, trophies 
+                        FROM players 
+                        WHERE season_id = $1 AND guild_id = $2
+                        ORDER BY defenses DESC
+                        LIMIT 15
+                    """
+        fetch = await ctx.db.fetch(query, season, ctx.guild.id)
+        defenses = {}
+        for row in fetch:
+            defenses[row['player_tag']] = row['defenses']
+        table = CLYTable()
+        table.title = f"Defense wins for Season {season}"
+        index = 0
+        async for player in self.bot.coc.get_players((n[0] for n in fetch)):
+            table.add_row([index, defenses[player.tag], player.trophies, player.name])
+        render = table.trophyboard_defenses()
+        fmt = render()
+
+        e = discord.Embed(colour=discord.Colour.dark_red(), description=fmt)
+        await ctx.send(embed=e)
+
+    @season_stats.command(name='gains', aliases=['trophies'])
+    async def season_stats_gains(self, ctx, season: typing.Optional[int] = None):
+        """Get trophy gains for all clans.
+
+           By default, you shouldn't need to call these sub-commands as the bot will
+           parse your argument and direct it to the correct sub-command automatically.
+
+           **Example**
+           :white_check_mark: `+season stats gains`
+        """
+        season = season or await self.bot.seasonconfig.get_season_id() - 1
+        query = """SELECT player_tag, trophies - start_trophies as gain, trophies 
+                            FROM players 
+                            WHERE season_id = $1 AND guild_id = $2
+                            ORDER BY gain DESC
+                            LIMIT 15
+                        """
+        fetch = await ctx.db.fetch(query, season, ctx.guild.id)
+        table = CLYTable()
+        table.title = f"Trophy Gains for Season {season}"
+        index = 0
+        for row in fetch:
+            player = await self.bot.coc.get_player(row['player_tag'])
+            table.add_row([index, row['gains'], player.trophies, player.name])
+        render = table.trophyboard_gain()
+        fmt = render()
+
+        e = discord.Embed(colour=discord.Colour.green(), description=fmt)
+        await ctx.send(embed=e)
+
+    @season_stats.command(name='donors', aliases=['donations', 'donates', 'donation'])
+    async def season_stats_donors(self, ctx, season: typing.Optional[int] = None):
+        """Get donations for all clans.
+
+           By default, you shouldn't need to call these sub-commands as the bot will
+           parse your argument and direct it to the correct sub-command automatically.
+
+           **Example**
+           :white_check_mark: `+season stats donations`
+        """
+        # TODO Since this is straight donations, might change sql to use donationevents (just want consistent numbers)
```

Comment:
This looks good as-is, this one should be perfectly true if we pull on time. I also like to keep queries down on that table because they take ages coz its massive (and I haven't sorted indexes properly)

Reply:
sounds good to me